src/twister2/plugin.py

```diff
@@ -34,6 +35,32 @@ def pytest_collect_file(parent, path):
         return YamlFile.from_parent(parent, path=Path(path))
 
 
+def pytest_generate_tests(metafunc: pytest.Metafunc):
+    # generate parametrized tests for each selected platform for ordinary pytest tests
+    if 'specification' not in metafunc.fixturenames:
+        return
+
+    twister_config = metafunc.config.twister_config
+
+    platforms_list: list[PlatformSpecification] = [
+        platform for platform in twister_config.platforms
+        if platform.identifier in twister_config.default_platforms
+    ]
+
+    params: list[pytest.param] = []
+    ids: list[str] = []
+    for platform in platforms_list:
+        params.append(
+            pytest.param(platform, marks=pytest.mark.platform(platform))
```

Comment:
You can pass `id` directly to `pytest.param` instead of creating separate list of `ids`, like this:
```python
pytest.param(platform, marks=pytest.mark.platform(platform), id=platform.identifier)
```
Thanks to this you can avoid risk that there will occur some discrepancy between list of params and ids in the future.

Reply:
Changed